src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -458,9 +433,8 @@ type TableContext<'TRecord> internal
     ///     Asynchronously fetches item of given key from table.
     /// </summary>
     /// <param name="key">Key of item to be fetched.</param>
-    /// <param name="collector">Function to receive request metrics.</param>
-    member __.GetItemAsync(key : TableKey, ?collector) : Async<'TRecord> = async {
-        let! item = getItemAsync key None collector
+    member __.GetItemAsync(key : TableKey) : Async<'TRecord> = async {
+        let! item = getItemAsync key None
         return template.OfAttributeValues item
```

Comment:
I guess this is for another PR, but getItemAsync could take a `render : IRecord -> 'T`, and that would allow this and the next to become one-liners without Async in the picture

Reply:
`getItemAsync key None |> Async.map template.OfAttributeValues` would also work, there’s a bunch of routine cleanup stuff that can be done at some point.

Reply:
...if we had `Async.map` in Core. Grr, why can’t we have nice things

Reply:
I personally frown on `Async.map` for style reasons (I very rarely want anything to be added to `FSharp.Core` in general, but especially for a library, I'd say this sort of  this is unwarranted (seealso [appeal to authority](https://eiriktsarpalis.wordpress.com/2017/04/02/programming-in-the-point-free-style/)))
- if there is no asynchrony intrinsically required I'd like it to be done inline given that there will be a needless Async layer involved.

Reply:
I definitely disagree on Async.bind / Async.map
But I appreciate the sentiment of wanting nice things (e.g. I want a [`Async.parallelThrottled`](https://github.com/fsprojects/FSharp.Control.AsyncSeq/issues/74#issuecomment-1042465992) but the same reasoning behind not having `Async.map` / `bind` is denying/protecting us ;)

Reply:
Hmm, probably a bit of a holy war, but personally I'd regard manually passing a synchronous callback/continuation just to avoid an async one here as making the code more obtuse - if it’s being done for performance I'd want to be really confident there’s a noticeable benefit. 

I have a feeling your motivation here though is wanting to try and define all the public methods as a single internal function call - I can certainly see the appeal, but there are much bigger offenders in the class.

Reply:
I'm certainly not a pointfree zealot, but I'll often use map/bind on various types to avoid (simple) CEs or `match` expressions. I acknowledge this may not be clearer for everyone but it is for me.

Reply:
Regarding Core, I’ve never understood why we have `map`/`bind` available for `Option`, `Result`, `List`, `Seq`, `Array`, `Set`, `Map` etc but not `Async`; it’s always seemed really inconsistent to me.

Reply:
In the general case I agree that it seems like a wart to not have Async.map and bind in the box

However, I would say that any time you can avoid triggering an Async or Task continuation, there are significant perf and GC benefits, which ultimately help scalability more than a microbenchmark will illustrate. IMO Having an `Async.map` in the box would trigger lots of use without appreciation for that concern.

And an admission: If it was in the box, I'd use it - I regularly think of it. But I have never declared one in an application helpers lib as I've seen its mere presence lead to horribly over-pipelined code and a failure to split effects from logic (i.e. if there is an Async.map and and Option.mapAsync in the picture, people don't end up writing code in keeping with the style espoused in Don's Code I Love talks)

I agree that passing a render function inward does make the code more obtuse - my main goal is making the `member` bodies as boring as possible, and `Async.map` / `Async.Ignore` vs passing a `render` function inward doesnt alter that equation much. I definitely would not go to the level of having any hard and fast rules about it though; I guess my heuristic is that I want that member body as terse as possible, but would not compromise perf to achieve that -- that's why I'd either pass a `render` function inward (less code, one less Async continuation) or leave it alone (make the `Async`ness explicit with an `async {}`)

Regarding `match` expressions, I tend to prefer them over `Option.bind`, `Option.filter`, `Option.exists`, `Option.defaultValue`, `Option.defaultWith` and the like as I (subjectively) find that less mindbending even if there are less chars.

Some of those I will use on occasion if it fits with the general work to be done in a given function, but I start with a `match` and refactor to operators, whereas I guess people coming from a more heavily functional background will take it from the exact opposite.  I'd go so far as to claim that `match` first is part of idiomatic F# a la F# code I Love Talks but I may be putting too many words in too many people's mouths and/or misremembering talks that I have not rewatched in recent times. (I was probably influenced in taking this view by having worked with Eirik who tended to follow a similar style in the codebases we worked on IIRC)

Can't beat a style debate eh ;)

Anyway thankfully this PR achieved the more important win of not having the metrics stuff having to be as in-your-face as I made it; thanks for making the time!